### PR TITLE
Move architecture-specific code out of codegencommon.cpp

### PR DESCRIPTION
**Pull request by BruceForstall**
Move about 3300 lines of architecture-specific code to the
architecture-specific codegen files (codegenarm.cpp, codegenarm64.cpp,
codegenarmarch.cpp, codegenxarch.cpp). Much of this is prolog/epilog
generation.

A new set of arch-specific functions, genZeroInitFrameUsingBlockInit,
is added to extract a huge amount of arch-specific code from genZeroInitFrame.

The moved code wasn't altered except to remove unnecessary ifdefs.

Minor:
- inlined and removed getAdjustSP
- removed unused Compiler versions of GetHasTailCalls/SetHasTailCalls

### CODE REVIEW

**Comment by BruceForstall** [comment]
@echesakovMSFT PTAL
@dotnet/jit-contrib

**Comment by BruceForstall** [comment]
cc @shushanhf

File: src/coreclr/jit/codegenxarch.cpp [hunk 1]
```diff
@@ -7368,7 +7449,14 @@ void CodeGen::genRemoveAlignmentAfterCall(GenTreeCall* call, unsigned bias)
 #else  // UNIX_X86_ABI
     if (bias != 0)
     {
-        genAdjustSP(bias);
+        if (bias == sizeof(int))
+        {
+            inst_RV(INS_pop, REG_ECX, TYP_INT);
```
Comment (echesakovMSFT) [inline src/coreclr/jit/codegenxarch.cpp:7454]: This can be further simplified to
```c++
if (bias == sizeof(int)) 
{
...
}
else if (bias != 0)
{
...
}
```

**Review by echesakov — APPROVED** [review]
Looks good - left one comment.

**Comment by AndyAyersMS** [comment]
I was going to say, maybe this could wait until I've finished this next round of changes for OSR....? Oh well.

**Comment by BruceForstall** [comment]
> Maybe you can look at [main...AndyAyersMS:OSRX64EpilogRedesign](https://github.com/dotnet/runtime/compare/main...AndyAyersMS:OSRX64EpilogRedesign) and give me some pointers on merging?

genPopCalleeSavedRegisters and genFnEpilog moved to codegenxarch.cpp. genFnProlog is still in codegencommon.cpp, so that should merge fine.

**Pull request merged**